portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -90,20 +92,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Take the input from the POST request 
-    String text = request.getParameter("text-input");
-    String author = request.getParameter("author");
-    // The added date for the comment will be the current date.
-    Date date = new Date();
-
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("text", text);
-    commentEntity.setProperty("author", author);
-    commentEntity.setProperty("date", date);
-
-    // Store comment data in the database
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentEntity);
+    // Check that the user is authenticated before adding the comment
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      // Take the input from the POST request 
+      String text = request.getParameter("text-input");
+      String email = userService.getCurrentUser().getEmail();
+      // The added date for the comment will be the current date.
+      Date date = new Date();
```

Comment:
It's better to keep related code together, e.g., it's better to create an object right before it's used.  In the case here, we can simply inline "new Date()" when you set the "date". 

BTW, what is the date format you get here? I'm asking because it's probably better to have a date with YY-MM-DD-SS.

The way the date is created has a problem. When it's created, there is no info about timezone. So readers couldn't really know  the date in their timezones. But this is just FYI, you don't need to fix it here.

Reply:
Thank you for the notice! The date appears like this in my page: "Jul 30, 2020 9:10:31 AM", so the day format seems fine, however you are right about the timezone - I don't know the timezone for the comments. I'll do my research about that.